File: src/nodes/compound.zig
const std = @import("std");
const nd = @import("node.zig");

pub const NodeCompound = struct {
    nodes: std.ArrayList(*nd.Node),

    pub fn init(nodes: std.ArrayList(*nd.Node)) NodeCompound {
        return NodeCompound{ .nodes = nodes };
    }

    pub fn get_nodes(self: *const NodeCompound) std.ArrayList(*nd.Node) {
        return self.nodes;
    }
};

File: src/nodes/compound.rs
use crate::nodes::node::Node;

pub struct NodeCompound {
    nodes: Vec<Node>,
}

impl NodeCompound {
    pub fn init(nodes: Vec<Node>) -> NodeCompound {
        NodeCompound { nodes }
    }

    pub fn get_nodes(&self) -> &Vec<Node> {
        &self.nodes
    }
}